src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -87,23 +83,16 @@ def _format_zinc_arguments(settings, distribution):
         :param settings: The jvm platform settings from which to extract the arguments.
         :type settings: :class:`JvmPlatformSettings`
         """
-        zinc_args = [
-            "-C-source",
-            f"-C{settings.source_level}",
-            "-C-target",
-            f"-C{settings.target_level}",
-        ]
-        if settings.args:
-            settings_args = settings.args
-            if any("$JAVA_HOME" in a for a in settings.args):
-                logger.debug(
-                    'Substituting "$JAVA_HOME" with "{}" in jvm-platform args.'.format(
-                        distribution.home
-                    )
-                )
-                settings_args = (a.replace("$JAVA_HOME", distribution.home) for a in settings.args)
-            zinc_args.extend(settings_args)
-        return zinc_args
+        args = list(settings.args)
+        # if source, target or release is already specified, then don't generate args for them.
+        if all(a not in ("-C-source", "-C-target", "-C--release") for a in args):
```

Comment:
Does this apply to any other args, i.e. should we avoid prefixing any `-C`-prefixed arg with an additional `-C`? does the pattern `-C-C` show up anywhere? If we avoid prefixing existing `-C` args with an additional `-C` it could be more general.

Reply:
The `-C` is specific to zinc. `-C` is passed to javac, `-S` is passed to scalac. Currently, pants only attempts to generate `-source` et al args for javac, so this is a simpler, tailored approach for zinc which is the meta compiler we're using. Arguably, I should add the check here to the javac compile task, but since this is for our usage and afaik no one else is using this branch, I focused on the compile chain we use without explicitly breaking other things.